packages/ses/src/module-load.js

```diff
@@ -0,0 +1,102 @@
+const { create, keys, values, freeze } = Object;
+
+// makeAlias constructs compartment specifier tuples for the `aliases` private
+// field of compartments.
+// These aliases allow a compartment to alias an internal module specifier to a
+// module specifier in an external compartment, and also to create internal
+// aliases.
+// Both are facilitated by the moduleMap Compartment constructor option.
+export const makeAlias = (compartment, specifier) =>
+  freeze({
+    compartment,
+    specifier,
+  });
+
+// Pre-compute resolutions of all imports within the compartment in which a
+// module was loaded.
+const resolveAll = (imports, resolveHook, fullReferrerSpecifier) => {
+  const resolvedImports = create(null);
+  for (const importSpecifier of keys(imports)) {
+    const fullSpecifier = resolveHook(importSpecifier, fullReferrerSpecifier);
+    resolvedImports[importSpecifier] = fullSpecifier;
+  }
+  return freeze(resolvedImports);
+};
+
+// load asynchronously loads ModuleStaticRecords and creates a complete graph
+// of ModuleCompartmentRecords.
+// The module records refer to each other by a reference to the dependency's
+// compartment and the specifier of the module within its own compartment.
+// This graph is then ready to be synchronously linked and executed.
+export const load = async (
```

Comment:
Checking my understanding: This code is only meant to support the asynchronous `import` and `importHook`. By itself, it is not yet a step towards supporting `importNow`. Is that correct?

Reply:
Load is indeed only used by `compartment.import`, not by `compartment.importNow`. Load captures the entire asynchronous portion of importing. It returns a promise and resolves that promise (with `undefined`) when one can call `importNow` for any module  in the closure of transitive dependencies.